### PR TITLE
ci: Reduce use of bash -c

### PR DESCRIPTION
**Pull request by maflcko**
It is confusing to treat commands as a single string. This change is
also required to support paths and strings with spaces in them in the
future.

### CODE REVIEW

File: ci/test/03_test_script.sh [hunk 1]
```diff
@@ -163,15 +163,16 @@ if [ -n "$USE_VALGRIND" ]; then
...
 
 if [ "$RUN_UNIT_TESTS_SEQUENTIAL" = "true" ]; then
-  bash -c "${TEST_RUNNER_ENV} DIR_UNIT_TEST_DATA=${DIR_UNIT_TEST_DATA} LD_LIBRARY_PATH=${DEPENDS_DIR}/${HOST}/lib ${BASE_OUTDIR}/bin/test_bitcoin --catch_system_errors=no -l test_suite"
+  DIR_UNIT_TEST_DATA="${DIR_UNIT_TEST_DATA}" LD_LIBRARY_PATH="${DEPENDS_DIR}/${HOST}/lib" "${BASE_OUTDIR}"/bin/test_bitcoin --catch_system_errors=no -l test_suite
 fi
 
 if [ "$RUN_FUNCTIONAL_TESTS" = "true" ]; then
-  bash -c "LD_LIBRARY_PATH=${DEPENDS_DIR}/${HOST}/lib ${TEST_RUNNER_ENV} test/functional/test_runner.py --ci $MAKEJOBS --tmpdirprefix ${BASE_SCRATCH_DIR}/test_runner/ --ansi --combinedlogslen=99999999 --timeout-factor=${TEST_RUNNER_TIMEOUT_FACTOR} ${TEST_RUNNER_EXTRA} --quiet --failfast"
+  # shellcheck disable=SC2086
+  LD_LIBRARY_PATH="${DEPENDS_DIR}/${HOST}/lib" test/functional/test_runner.py --ci "${MAKEJOBS}" --tmpdirprefix "${BASE_SCRATCH_DIR}"/test_runner/ --ansi --combinedlogslen=99999999 --timeout-factor="${TEST_RUNNER_TIMEOUT_FACTOR}" ${TEST_RUNNER_EXTRA} --quiet --failfast
```
Comment (vasild) [inline ci/test/03_test_script.sh:175]: nit: some inconsistency wrt `"${FOO}"/bar` vs `"${FOO}/bar"` which to my understanding are the same thing.

File: ci/test/00_setup_env_i686_multiprocess.sh [hunk 2]
```diff
@@ -14,4 +14,4 @@ export DEP_OPTS="DEBUG=1 MULTIPROCESS=1"
 export GOAL="install"
 export BITCOIN_CONFIG="--enable-debug CC='clang -m32' CXX='clang++ -m32' \
 LDFLAGS='--rtlib=compiler-rt -lgcc_s' CPPFLAGS='-DBOOST_MULTI_INDEX_ENABLE_SAFE_MODE'"
-export TEST_RUNNER_ENV="BITCOIND=bitcoin-node"
```
Comment (vasild) [inline ci/test/00_setup_env_i686_multiprocess.sh:17]: `ci/test/00_setup_env.sh` still contains `export TEST_RUNNER_ENV=${TEST_RUNNER_ENV:-}` which can be dropped as well coz it is the only remaining reference to `TEST_RUNNER_ENV`.

**Review by vasild — APPROVED** [review]
ACK fad0165de85e1c0a39e511a619e9efcee69ac581 modulo the removal of the now stale reference to `TEST_RUNNER_ENV`.

**Review by vasild — APPROVED** [review]
ACK fad82fea2bef88c9f11e25ca43c7886a2b9b5da2

**Comment by RandyMcMillan** [comment]
utACK fad82fea2bef88c9f11e25ca43c7886a2b9b5da2

**Pull request merged**